en/concept/flight_modes.md

```diff
@@ -1,111 +1,215 @@
 - [Drive Modes (Ackermann Rover)](../flight_modes_rover/ackermann.md)
+- [Basic Configuration > Flight Modes](../config/flight_mode.md)
+
+::: info
+At time of writing there are no PX4 ROS 2 flight modes that are considered "supported as an integral part of the PX4 open source offering".
+:::
+
+## PX4 FC Modes vs PX4 ROS2 Flight Modes
+
+With some exceptions it is possible to write a mode in either the FC and companion computer.
+The main considerations are listed below.
+
+ROS 2 modes cannot be used in the following cases:
+
+- Modes that need to run on vehicles that don't have a companion computer.
+- Modes that require low-level access, strict timing, and/or high update rate requirements.
+  For example, a multicopter mode that implements direct motor control.
+- Safety critical modes, such as [Return mode](../flight_modes_mc/return.md).
+- When you can't use ROS (for any reason).
+
+PX4 ROS 2 modes should be considered for all other cases.
+They have the following benefits:
+
+- Easier to implement as there is no need to deal with low-level embedded constraints and requirements (such as restricted stack sizes).
+- Easier to maintain as the integration API is small, well defined, and stable (porting custom PX4 modes on the flight controller between PX4 versions is much harder).
+- Crashing the app does not crash the vehicle.
+- They can override existing modes to provide more advanced features.
+  You can even override a safety-critical mode with a better versions: if the ROS 2 mode crashes the original mode will be engaged.
+- High-level functionality is available, including a better-feature programming environment, and many useful Linux and ROS libraries.
+- More available compute to do more advanced processing (e.g. computer vision).
+
+Note that there are some negatives to keep in mind:
+
+- There is no CI environment for testing ROS 2 applications with PX4 in CI (it is easier to automate testing of PX4 flight modes running on the FC).
+- The [PX4 ROS 2 Control Interface](../ros2/px4_ros2_control_interface.md) that is used to define the modes is experimental
+- There is no formal "PX4 companion computer flight stack", so there is nowhere to contribute PX4 ROS2 modes such that they are become a part of the standard offering available to all users.
+
+## PX4 ROS 2 Flight Modes (Companion)
+
+[PX4 ROS 2 Control Interface](../ros2/px4_ros2_control_interface.md) documents how to create flight modes running on a companion computer.
 
+## PX Flight Modes (FC)
+
+The specific control behaviour of a flight mode at any time is determined by a [Flight Task](../concept/flight_tasks.md).
+A flight mode might define one or more tasks that define variations of the mode behavior.
+The actual behaviour that is used is normally defined in a parameter, and selected in [src/modules/flight_mode_manager/FlightModeManager.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/flight_mode_manager/FlightModeManager.cpp#L266-L285)
+
```

Comment:
@MaEtUgR Can I get some hints for flight modes. It looks like the bit of the code that does something is the Flight Task, and that a mode might have a few of these all that do a particular variant of the mode behaviour - e.g. different types of position control.

I also understand the mode restrictions, which is in the next section (I think).

But not clear 
- how I declare an entirely new mode "HamishMode".
- Where I have to add the code for that.
- Whether I have to add that to the list of modes exported to MAVLink or that happens magically
- If there are any base classes - i.e. What is a mode.

Any chance of some bullet points of the key concepts?